lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -1336,23 +1335,23 @@ namespace Js
         } \
         return DynamicObject::DeleteProperty(propertyNameString, flags);
 
-        if (BuiltInPropertyRecords::lastIndex.Equals(propertyName))
+        if (BuiltInPropertyRecords::lastIndex.Equals(propertyNameString))
         {
             DELETE_PROPERTY(true);
         }
-        else if (BuiltInPropertyRecords::global.Equals(propertyName)
```

Comment:
Since the one `propertyNameString` is used so often here, would it make sense to convert to a `PropertyRecord` as you did for another large chunk of comparisons?

Reply:
The other place, most of it shares the same length.Here, the lengths are different. It's not going to scan all of them no matter what.

Reply:
Won't it still try to construct a `LiteralStringWithPropertyStrPtr` (if not now, after the other PR goes in) before doing a length check? I would expect that  to have some overhead.

Reply:
If we will never use that string, right! otherwise, we will eventually fill that in.

Reply:
Used new GetPropertyRecord instead.